### PR TITLE
PayPal on checkout

### PR DESCRIPTION
**Pull request by jamesgorrie**
Part of #5722

### CODE REVIEW

File: support-frontend/assets/pages/[countryGroupId]/checkout.tsx [hunk 1]
```diff
@@ -316,6 +346,8 @@ export function Checkout() {
 	const [billingPostcodeStateLoading, setBillingPostcodeStateLoading] =
 		useState(false);
 
+	const formRef = useRef<HTMLFormElement>(null);
+	const [, setIsFormValid] = useState(false);
```
Comment (GHaberis) [inline support-frontend/assets/pages/[countryGroupId]/checkout.tsx:350]: What's going on here? I've not seen a use case for `useState` like this before 🤔 ...where we don't define a variable, such as `isFormValid` as the first item in destructured array.

Comment (jamesgorrie) [reply support-frontend/assets/pages/[countryGroupId]/checkout.tsx]: Good shout, we don't use this. I was trying to find a way to get the `disable`/`enable` methods from PayPal - [but found a better way](https://github.com/guardian/support-frontend/pull/5875/files#diff-dbc5d74a5f52e0cc93a8185ea2825653897711f50a6af3029dd91a90bcdd145fR795-R806).

File: support-frontend/assets/pages/[countryGroupId]/checkout.tsx [hunk 2]
```diff
@@ -785,14 +831,110 @@ export function Checkout() {
...
+													enable();
+												} else {
+													disable();
+												}
+											});
+										}}
+										funding={{
+											disallowed: [window.paypal.FUNDING.CREDIT],
+										}}
+										onClick={() => {
```
[the remark this answers is not in the document]
Comment (jamesgorrie) [reply support-frontend/assets/pages/[countryGroupId]/checkout.tsx:926]: I think we might need to do some tracking (as tracked in #5722)

File: support-frontend/assets/pages/[countryGroupId]/checkout.tsx [hunk 3]
```diff
@@ -442,6 +475,19 @@ export function Checkout() {
 											}
 										});
 								}
+								if (paymentMethod === 'PayPal') {
+									const paymentFields = {
+										recaptchaToken: '',
+										paymentMethod: 'PayPal',
+										baid: formData.get('payPalBAID') as string,
+									};
+									console.info('Posting data', {
```
[the remark this answers is not in the document]
Comment (jamesgorrie) [reply support-frontend/assets/pages/[countryGroupId]/checkout.tsx]: Removed and it now posts to `/create`.

**Review by GHaberis — APPROVED** [review]
Looks good 👍

**Pull request merged**